adagrams/game.py

```diff
@@ -1,13 +1,92 @@
 from random import randint
 
+LETTER_POOL_DISTRIBUTION = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2,
+    'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2,
+    'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1,
+    'Y': 2, 'Z': 1
+}
+
+SCORES = {
+        "A": 1, "E": 1, "I": 1, "O": 1, "U": 1, "L": 1, "N": 1, "R": 1, "S": 1, "T": 1,
+    "D": 2, "G": 2, "B": 3, "C": 3, "M": 3, "P": 3, "F": 4, "H": 4, "V": 4, "W": 4, "Y": 4,
+    "K": 5, "J": 8, "X": 8, "Q": 10, "Z": 10
+    }
+
+LETTER_POOL_LIST = []
+for letter, quantity in LETTER_POOL_DISTRIBUTION.items():
+    for _ in range (quantity):
+        LETTER_POOL_LIST.append(letter) 
+
+
 def draw_letters():
-    pass
+    copy_list = list(LETTER_POOL_LIST)
+    hand = []       
+    hand_length = 10
+        
+    for _ in range(hand_length):
+        current_len = len(copy_list)
+        random_index = randint(0, current_len - 1)
+        letter = copy_list.pop(random_index)
```

Comment:
Recall from our Lists and Memory session, that `.pop` has an `O(n)` time complexity when provided an index. This could lead to quadratic, or higher, time complexities when used in loops. Though, we don't run that risk here since our outer loop runs a constant number of times (10), what how could we refactor this avoiding the use of `.pop` and its linear complexity?